File: 4. Structures/bee-1911.py
n = int(input())
while n != 0:
    attend = {}
    for i in range(n):
        data = input().split()
        attend[data[0]] = data[1]
    m = int(input())
    cnt = 0
    for i in range(m):
        data = input().split()
        if attend[data[0]] != data[1]:
            x, y = [attend[data[0], data[1]]]
            cnt2 = 0
            for j in range(len(x)):
                if x[j] != y[j]:
                    cnt2 += 1

            cnt += 1 if cnt2 > 1 else 0
    print(cnt)
    n = int(input())

File: 4. Structures/bee-1911.go
package main

import (
	"fmt"
)

func main() {
	var (
		n, m int
		k, v string
	)
	fmt.Scan(&n)
	for n != 0 {
		attend := make(map[string]string)
		for i := 0; i < n; i++ {
			fmt.Scan(&k, &v)
			attend[k] = v
		}
		fmt.Scan(&m)
		cnt := 0
		for i := 0; i < m; i++ {
			fmt.Scan(&k, &v)
			if attend[k] != v {
				x, y := attend[k], v
				cnt2 := 0
				for j := 0; j < len(x); j++ {
					if string(x[j]) != string(y[j]) {
						cnt2++
					}
				}
				if cnt2 > 1 {
					cnt += 1
				}
			}
		}
		fmt.Println(cnt)
		fmt.Scan(&n)
	}
}
